fxrunner/src/lib/request.rs

```diff
@@ -0,0 +1,176 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::borrow::Cow;
+use std::io;
+use std::path::{Path, PathBuf};
+
+use async_trait::async_trait;
+use rand::prelude::*;
+use thiserror::Error;
+use tokio::fs::create_dir;
+
+use crate::fs::PathExt;
+
+#[derive(Clone)]
+pub struct RequestInfo<'a> {
+    pub id: Cow<'a, str>,
+    pub path: PathBuf,
+}
+/// A trait for creating and validating requests.
+#[async_trait]
+pub trait RequestManager {
```

Comment:
I think "request" may be a bit vague or even misleading for all of this. Maybe something like "trial" or "session" or "transaction". This doesn't need to change now, but it's definitely confusing coming in.